tests/rbd/test_rbd_groups_image_usecase.py

```diff
@@ -0,0 +1,102 @@
+from ceph.rbd.initial_config import initial_rbd_config
+from ceph.rbd.utils import getdict
+from ceph.rbd.workflows.cleanup import cleanup
+from ceph.rbd.workflows.group import (
+    add_image_to_group_and_verify,
+    create_group_and_verify,
+)
+from ceph.rbd.workflows.rbd import wrapper_for_image_ops
+from ceph.utils import get_node_by_id
+from cli.rbd.rbd import Rbd
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_group_creation_images_add(rbd_obj, client, **kw):
+    """
+    Tests the group creation followed by adding images and IO On them
+    Args:
+        client: rbd client obj
+        **kw: test data
+    """
+    kw["client"] = client
+    rbd = rbd_obj.get("rbd")
+
+    for pool_type in rbd_obj.get("pool_types"):
+        rbd_config = kw.get("config", {}).get(pool_type, {})
+        multi_pool_config = getdict(rbd_config)
+        for pool, pool_config in multi_pool_config.items():
+            if "data_pool" in pool_config.keys():
+                _ = pool_config.pop("data_pool")
+
+            # 1. create a group per pool
+            group = kw.get("config", {}).get("group", "image_group_default")
+            group_create_kw = {"client": client, "pool": pool, "group": group}
+            rc = create_group_and_verify(**group_create_kw)
+            if rc:
+                log.error(f"group {group} creation failed, test case fail")
+                return 1
+            else:
+                log.info("STAGE: group creation succeeded")
+
+            # 2. Add add images to the group
+            for image, image_config in pool_config.items():
+                add_image_group_kw = {
+                    "client": client,
+                    "pool": pool,
+                    "group": group,
+                    "image": image,
+                }
+                rc = add_image_to_group_and_verify(**add_image_group_kw)
+                if rc:
+                    log.error(f"Image {image} add failed, test case fail")
+                    return 1
+                else:
+                    log.info(f"STAGE: image {image} add succeeded")
+
+            # 3. Running IO on the images added to the group
```

Comment:
can we cover some snap and clone to that group images and perform some RBD operations like image resize, remove, etc.

Reply:
while there are lot many things we could do, currently in this sprint the focus is on 2 usecases, as we discussed we are planning to autmate only 2 usecases, not all functional combinations. So let this be there for 2 customer usecases. SInce not all customers may not do group snapshot I added "group image add" and "group snap" to loosely decouple

1. group image add
2. group snap